File: src/main/java/com/llacerximo/booklist/domain/model/Book.java
package com.llacerximo.booklist.domain.model;

import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {

    Long id;
    String title;
    Integer pages;
    Integer publishYear;
    Double sagaNum;
    Boolean fave;
    String image;
    Saga saga;
    Pseudonym author;
    List<Reread> rereads;
    List<Genre> genres;

}

File: web/src/app/components/books/book.ts
import {Saga} from "../sagas/saga";
import {Author} from "../authors/author";
import {Genre} from "../genres/genre";
import {Reread} from "./reread/reread";

export interface Book {

  id?: number;
  title: string;
  pages: number;
  publishYear: number;
  sagaNum?: number;
  fave: boolean;
  image: string;
  saga?: Saga;
  rereads: Reread[];
  authors: Author[];
  genres: Genre[];
  lastReread: Reread;
  readTime: number;
}
